dagshub/common/api/repo.py

```diff
@@ -259,78 +260,134 @@ def get_storage_file(self, path: str) -> bytes:
             raise RuntimeError(error_msg)
         return res.content
 
-    def _get_files_in_path(self, path, revision=None, recursive=False) -> List[ContentAPIEntry]:
+    def _get_files_in_path(
+        self, path, revision=None, recursive=False, traverse_storages=False
+    ) -> List[ContentAPIEntry]:
         """
         Walks through the path of the repo, returning non-dir entries
         """
 
-        dir_queue = [path]
+        dir_queue = []
         files = []
 
+        list_fn_folder = partial(self.list_path, revision=revision)
+        list_fn_storage = self.list_storage_path
+
+        def push_folder(folder_path):
+            dir_queue.append((folder_path, list_fn_folder))
+
+        def push_storage(storage_path):
+            dir_queue.append((storage_path, list_fn_storage))
+
+        # Initialize the queue
+        is_storage_path = str(path).lstrip("/").split("/")[0] in {"s3:", "gs:", "azure:"}
+        if is_storage_path:
```

Comment:
If the root is a storage path, I think you have to set `traverse_storages=True` to correctly handle the recursion

Reply:
`traverse_storages` is False ONLY when the root is being traversed (in the caller of this function).

Basically I wanted to make it so users don't need to explicitly set `traverse_storages=True` when downloading `s3://my-bucket`, because I think that'd be a surprising behavior for them (why do i have to flip this flag when I'm downloading a bucket folder but not a repo folder)